bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/SimpleDeviceHandler.java

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    /**
+     * Process the command for this thing.
+     *
+     * Only refresh is supported in this case.
+     *
+     * @param channelUID channel for which the command was issued
+     * @param command to execute
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        @Nullable
+        final EnergyManagerHandler bridgeHandler = this.getEnergyManagerHandler();
+        if (bridgeHandler != null) {
+            if (command instanceof RefreshType) {
+                this.logger.debug("update command for {}", this.getThing().getUID());
+                this.updateStatus(ThingStatus.ONLINE);
+                this.updateDeviceProperties();
+                this.updateDeviceChannels();
+            }
+        } else {
+            this.logger.error("Thing {} has no bridgeHandler for Bridge {}", this.getThing().getUID(),
+                    this.getThing().getBridgeUID());
+        }
+    }
+
+    /**
+     * Update the state of all channels.
+     */
+    protected void updateDeviceChannels() {
+        this.logger.trace("updateDeviceChannels for {}", this.getThing().getUID());
+        // find device for the thing
+        @Nullable
+        Device device = this.getDevice();
+
+        if (device != null) {
+            device.getStateValues().forEach((stateName, stateValue) -> {
+                this.logger.trace("{}: {} - {}", this.getThing().getUID(), stateName, stateValue.toFullString());
+                this.updateState(stateName, stateValue);
+            });
+        }
+    }
+
+    /**
+     * Assert that all {@link org.openhab.core.thing.type.ChannelType}s are registered for this thing.
+     */
+    protected void initDeviceChannels() {
+        this.logger.trace("initDeviceChannels for {}", this.getThing().getUID());
+        // find device for the thing
+        @Nullable
+        Device device = this.getDevice();
+
+        if (device != null) {
+            device.getSolarwattChannelSet().forEach((channelTag, solarwattChannel) -> {
+                this.logger.trace("{}: {}", this.getThing().getUID(), solarwattChannel.getChannelName());
+                this.assertChannel(solarwattChannel);
+            });
+        }
+    }
+
+    /**
+     * Update the properties for this device.
+     */
+    protected void updateDeviceProperties() {
+        this.logger.trace("updateDeviceProperties for {}", this.getThing().getUID());
+        // find device for the thing
+        @Nullable
+        Device device = this.getDevice();
+
+        if (device != null) {
+            // update properties
+            Map<String, String> properties = this.editProperties();
+            this.putProperty(properties, PROPERTY_ID_NAME, device.getIdName());
+            this.putProperty(properties, PROPERTY_ID_FIRMWARE, device.getIdFirmware());
+            this.putProperty(properties, PROPERTY_ID_MANUFACTURER, device.getIdManufacturer());
```

Comment:
There are predefined system constants, you could use. For example:
```suggestion
            this.putProperty(properties, Thing.PROPERTY_VENDOR, device.getIdManufacturer());
```

Reply:
These are string if the form `IdName`, `IdFirmware` and `IdManufacturer`. These are supplied with that exact naming from the energymanager.

Reply:
OK